src/x86/avx.rs

```diff
@@ -85,13 +85,78 @@ pub fn _mm256_floor_pd(a: f64x4) -> f64x4 {
     unsafe { roundpd256(a, 0x01) }
 }
 
+/// Round packed single-precision (32-bit) floating point elements in `a`
+/// according to the flag `b`. The value of `b` may be as follows:
+/// 0x00: Round to the nearest whole number.
+/// 0x01: Round down, toward negative infinity.
+/// 0x02: Round up, toward positive infinity.
+/// 0x03: Truncate the values.
+/// For a few additional values options, check the LLVM docs:
+/// https://github.com/llvm-mirror/clang/blob/dcd8d797b20291f1a6b3e0ddda085aa2bbb382a8/lib/Headers/avxintrin.h#L382
+#[inline(always)]
+#[target_feature = "+avx"]
+// #[cfg_attr(test, assert_instr(vroundps))]
```

Comment:
I think this can be tested through the make-a-shim-that-calls-this-with-an-immediate strategy, right?

Reply:
I couldn't get it working reliably for round_pd, so I'm leaving it off of round_ps for now. I think the stuff in mod tests should cover us until we find a better way of doing it.

Reply:
Does this work?

```rust
#[cfg(test)]
#[cfg_attr(test, assert_instr(vroundps))]
fn test_mm256_round_ps(a: f32x8) -> f32x8 {
    _mm256_found_ps(a, 3)
}
```

Reply:
That's the ticket. Just added it for round_*.